### PR TITLE
feat: React root including MainLayout, useBAINotification

### PR DESCRIPTION
**Pull request by yomybaby**
This PR modifies the structure to have React as the root, with the existing backend.ai-webui components nested within. This change includes managing the Main layout, which corresponds to the Sider and Menu, within React.

```[tasklist]
### Feature
- [x] config options related to Sider menus (such as [menu_blocklist](https://github.com/lablup/backend.ai/pull/1727), [menu_inactivelist](https://github.com/lablup/backend.ai/pull/1733)...)
- [x] Documentation link on header ([#9f757](https://github.com/lablup/backend.ai-webui/pull/2121/commits/9f75721b9f951e04f333d403d6c42388b293f749), [#62781](https://github.com/lablup/backend.ai-webui/pull/2121/commits/62781c7141c1856a5dae22d9a2220cd763538adf))
- [x] Notification list
- [x] Task list
- [x] Plugin menu ([#3f3e1](https://github.com/lablup/backend.ai-webui/pull/2121/commits/3f3e11fa28e734f181ac8491377b681a5d67fec6))
- [x] Custom theme for sider configuration ([#f6bf4](https://github.com/lablup/backend.ai-webui/pull/2164/commits/f6bf42607ee585b525a47f3abe2229b900b24d1f))
- [x] ~~Apply light bg color to mwc-tab-bar~~ => https://github.com/lablup/backend.ai-webui/issues/2160
- [x] Collapse sidebar config (`Use Compact Sidebar by default`) ([#2288c](https://github.com/lablup/backend.ai-webui/pull/2121/commits/2288ccfd3d0191d05b1c3cfb760d072988e321bc))
- [x] Add notification / task filtering
- [x] 404 page
- [x] ~~routing 401, 404 and plugin pages in React part (react-router redirect)~~ => https://github.com/lablup/backend.ai-webui/issues/2160
- [x] ~~Timeout & off-line notification~~ => https://github.com/lablup/backend.ai-webui/issues/2160
- [x] Display/Hide menu item on Sider depends on manage version (sujin) ([#bd94355](https://github.com/lablup/backend.ai-webui/pull/2121/commits/bd94355b8bfc0e5c4f18a56bd169339eb83931fc))
```

```[tasklist]
### UI/Layout changes and bugs
- [x] Add padding to the notification title (or close button) so that there is no overlapping.
- [x] `bg_task_failed` or `task_failed` of background-task SSE? (handle both)
- [x] ~~When initial rendering, the badge of Notification Button is flickering~~ => https://github.com/lablup/backend.ai-webui/issues/2160
- [x] potential undefined in plugin menu items when no plugins (you can test try to open 404 page eg`/agent1231312`) [commit](https://github.com/lablup/backend.ai-webui/pull/2121/commits/5294538401cdccd0e2dce112b37bf847cb0f24db)
- [x] Missing: react route for Serving detail page (jongeun) ([commit](https://github.com/lablup/backend.ai-webui/pull/2121/commits/d01531cfbc807425b54a9613fe08cc1589d0daad))
- [x] Bug: No notification when login failed ([ref](https://teams.microsoft.com/l/message/19:14c484402d874dafb15806d093b95a82@thread.skype/1705991124356?tenantId=13c6a44d-9b52-4b9e-aa34-0513ee7131f2&groupId=74ae2c4d-ec4d-4fdf-b2c2-f5041d1e8631&parentMessageId=1705990885007&teamName=devops&channelName=Frontend&createdTime=1705991124356)) (jongeun) ([commit](https://github.com/lablup/backend.ai-webui/pull/2121/commits/3872e2f2e4f0527c60119d37356d9be78be103f3))
- [x] Theme: apply light theme to lit tab ([commit](https://github.com/lablup/backend.ai-webui/pull/2121/commits/c9b4f8c1931710d100eaff499ed902c2086f0bfa)) (sujin)
- [x] Display functions according to the settings in config.toml.(e.g. userSignout, siteDescription, etc..) ([commit](https://github.com/lablup/backend.ai-webui/pull/2121/commits/042c7c77787db6fdd79d2aa028eacc5051ca925e))
- [x] Bug: "자원사용량", "시스템 자원" broken UI ([ref](https://teams.microsoft.com/l/message/19:14c484402d874dafb15806d093b95a82@thread.skype/1705991052693?tenantId=13c6a44d-9b52-4b9e-aa34-0513ee7131f2&groupId=74ae2c4d-ec4d-4fdf-b2c2-f5041d1e8631&parentMessageId=1705990885007&teamName=devops&channelName=Frontend&createdTime=1705991052693)) @agatha197 ([commit](https://github.com/lablup/backend.ai-webui/pull/2121/commits/8852e8896391d65bee3ebb9c6a0567082ad4171d))
- [x] Bug: Broken layout of Import page ([ref](https://teams.microsoft.com/l/message/19:14c484402d874dafb15806d093b95a82@thread.skype/1705991318489?tenantId=13c6a44d-9b52-4b9e-aa34-0513ee7131f2&groupId=74ae2c4d-ec4d-4fdf-b2c2-f5041d1e8631&parentMessageId=1705990885007&teamName=devops&channelName=Frontend&createdTime=1705991318489), jongeun, [commit](https://github.com/lablup/backend.ai-webui/pull/2121/commits/f8b22adf3fbe80d9d3036716ca278ea4e3e1ee5f))
- [x] Bug: userDropdown is broken UI (jongeun, [commit](https://github.com/lablup/backend.ai-webui/pull/2121/commits/3b8e2064c418b4e83a5ca6a10af6c4d62704a2e4))
- [x] ~~Bug: user can navigate to use dropdown menu or webui logo even though page is added in blocklist and inactivelist @agatha197~~ => https://github.com/lablup/backend.ai-webui/issues/2160
- [x] Bug: Notification is not shown correct ([ref](https://github.com/lablup/backend.ai-webui/pull/2121#issuecomment-1907405309)) (jongeun, [commit](https://github.com/lablup/backend.ai-webui/pull/2121/commits/a6dfd3d4161155dcbc5b81f3f4f6f58e9359951b))
- [x] Bug: notification drawer over the mask of lit modal (jongeun, [commit](https://github.com/lablup/backend.ai-webui/pull/2121/commits/18931a9d07906e6dc89cab028a4ea8e35274b0a3))
- [x] Bug: (electron) Slider menu 'summary' item doesn't selected when login. ([ref](https://github.com/lablup/backend.ai-webui/pull/2121#issuecomment-1907567470)) (jongeun, [commit](https://github.com/lablup/backend.ai-webui/pull/2121/commits/08a610ef354625e4e77555f9884c96b4f4132f37))
- [x] Bug: (electron) Unexpected Application Error on Logout ([ref](https://github.com/lablup/backend.ai-webui/pull/2121#pullrequestreview-1840731380)) (jongeun, [commit](https://github.com/lablup/backend.ai-webui/pull/2121/commits/5f9bce632aa8a714e62670410e8c82ae27669c85))
- [x] ~~Bug: (electron) When using TableColumnsSettingModal, selected changes not applied. ([ref](https://github.com/lablup/backend.ai-webui/pull/2121#pullrequestreview-1840828015)) (jongeun)~~ handled in https://github.com/lablup/backend.ai-webui/pull/2178
- [x] Bug: ~~the click event doesn't work after clicking the `create a new keypair` button of the summary page. (superadmin only)~~ -> handle in https://github.com/lablup/backend.ai-webui/issues/2176
- [x] Bug: error notification's detail button in drawer doesn't navigate to log page (jonguen, [commit](https://github.com/lablup/backend.ai-webui/pull/2121/commits/bd07f8149e31f1db869dd9b78af6646a6c7424c6))
```

```[tasklist]
### UI/Layout Idea
- [x] Compact sider setting [2288ccf](https://github.com/lablup/backend.ai-webui/pull/2121/commits/2288ccfd3d0191d05b1c3cfb760d072988e321bc)
- [x] Drawer UX: Not overlay style and change the close control (applied "drawer on content" style)
- [x] Reduce the Sider width
```

## How to test
- Run running script again so that recognize the renamed file, react/src/hooks/index.**tsx** (`npm run server:d`, `npm run build:d`) 

## Reviewer check list
Please ref [loop page](https://lablup.sharepoint.com/:fl:/s/fb28b4ec-f199-4e97-a2b6-c73a1fcdd1ff/EZewmhUKvrdMpsFGKPJsbuMB2N5DRcRpS7AbQqBUHHnsNA?e=1sFCmk&nav=cz0lMkZzaXRlcyUyRmZiMjhiNGVjLWYxOTktNGU5Ny1hMmI2LWM3M2ExZmNkZDFmZiZkPWIlMjFlS3Z6RE5UdHUwbWxLRVJwTEhPeWtibHFzYU12SDlkS2xLeTZHa29tcVUyUTBZcTdyUFJDVEpZdkxqVktWU2h3JmY9MDFXUkVWVkRNWFdDTkJLQ1Y2VzVHS05RS0dGRFpHWTNYRCZjPSUyRiZhPUxvb3BBcHAmcD0lNDBmbHVpZHglMkZsb29wLXBhZ2UtY29udGFpbmVyJng9JTdCJTIydyUyMiUzQSUyMlQwUlRVSHhzWVdKc2RYQXVjMmhoY21Wd2IybHVkQzVqYjIxOFlpRmxTM1o2UkU1VWRIVXdiV3hMUlZKd1RFaFBlV3RpYkhGellVMTJTRGxrUzJ4TGVUWkhhMjl0Y1ZVeVVUQlpjVGR5VUZKRFZFcFpka3hxVmt0V1UyaDNmREF4VjFKRlZsWkVTVmd5VEVaSVRVNWFTelpPUkVsTVUwVk1XRk5DUmtKTVFqVSUzRCUyMiUyQyUyMmklMjIlM0ElMjIzOTUyZDY0NC00NDI3LTQzZWUtYjJhMS05YzE0OWY0NjY2MzElMjIlN0Q%3D). 

**Checklist:** (if applicable)

- [ ] Mention to the original issue
- [ ] Documentation
- [ ] Minium required manager version
- [x] Specific setting for review (eg., KB link, endpoint or how to setup)
- [x] Minimum requirements to check during review
- [ ] Test case(s) to demonstrate the difference of before/after

### CODE REVIEW

**Comment by yomybaby** [comment]
👍 @agatha197  I think we are almost finished. 
We are making many changes in this PR, so we need to test it carefully.
Please add a PR checkpoint to the PR description for the reviewer.

**Comment by yomybaby** [comment]
~~When i18n language is changed, the page title on the Header wasn't changed. (I'll fix this)~~ FIXED.
image

**Comment by yomybaby** [comment]
Push commits:
- Reduce the sider width 
- Fix slider collapsing condition

**Comment by yomybaby** [comment]
Push commits:
- Reduce the sider width 
- Fix slider collapsing condition

**Comment by yomybaby** [comment]
Oops. The serving detail page is missing. I'll fix it.

**Review by gee05053 — CHANGES_REQUESTED** [review]
스크린샷 2024-01-23 오후 5 44 29
스크린샷 2024-01-23 오후 5 50 47

@yomybaby 
tab's color is strange.

**Comment by agatha197** [comment]
> 스크린샷 2024-01-23 오후 5 44 29 스크린샷 2024-01-23 오후 5 50 47
> 
> @yomybaby tab's color is strange.

oops, my mistake. I'll handle it.

File: src/components/backend-ai-maintenance-view.ts [hunk 1]
```diff
@@ -190,61 +190,187 @@ export default class BackendAiMaintenanceView extends BackendAIPage {
...
-          if (err && err.message) {
-            this.notification.text = PainKiller.relieve(err.title);
-            this.notification.detail = err.message;
-            this.notification.show(true, err);
-          }
-        }),
-      '',
-      'image',
-    );
+    let taskId;
```
Comment (gee05053) [inline src/components/backend-ai-maintenance-view.ts]: @agatha197 taskId is unused variable

**Review by gee05053 — CHANGES_REQUESTED** [review]
@agatha197 I added to tasklist.

Bug: user can navigate to use dropdown menu or webui logo even though page is added in blocklist and inactivelist

**Review by ironAiken2 — CHANGES_REQUESTED** [review]
On the Agent Summary page, the UI changes for each network request.
1. After resizing the cell, the size will change randomly as the network request progresses.
2. The phrase "No Agents to display" will be re-rendered on each network request.

Additionally, the default cell width is different when there is a list value and when it is empty.

image
image

image

**Comment by gee05053** [comment]
@yomybaby 
After an error occur, notification is not shown correct content

스크린샷 2024-01-24 오후 2 15 02

**Comment by gee05053** [comment]
@yomybaby @agatha197 
only electron occur.
when user login and summary page is appeared, sidebar is not select.
스크린샷 2024-01-24 오후 4 35 49

**Review by ironAiken2 — CHANGES_REQUESTED** [review]
Only occurs in an electron environment. An error occurs on logout.

@yomybaby @agatha197 
image

**Review by ironAiken2 — CHANGES_REQUESTED** [review]
Only occurs in an electron environment. 

If you customize the filtering criteria in TableColumnsSettingModal and press OK, the data will be filtered correctly, but when you reopen the modal, the selected criteria will not be applied. If you reopen the modal one more, then will be applied.

@yomybaby @agatha197 

image
image

**Review by gee05053 — CHANGES_REQUESTED** [review]
@yomybaby 
I found issue after you added detail button in error notification.
It work correctly at error notification.
But if you click error notification's detail button in notification drawer, you navigate to `usersetting` page
스크린샷 2024-01-26 오후 3 03 37
스크린샷 2024-01-26 오후 3 04 13

**Review by gee05053 — APPROVED** [review]
LGTM😊

**Pull request merged**